### PR TITLE
Azure - Add Metrics Config Option & Update Application Insights & Metrics Docs

### PR DESCRIPTION
**Pull request by aluong**
- Updated the Azure Getting Started monitoring section
- Provide more details on the Application Insights logs and metrics

### Preview Rendering
https://custodian8bf14897.blob.core.windows.net/andy-docs/html/azure/advanced/appinsightslogging.html

### CODE REVIEW

File: docs/source/azure/gettingstarted.rst [hunk 1]
```diff
@@ -90,17 +90,8 @@ See :ref:`filters` for more information on the features of the Value filter used
...
-  $ custodian run -s <output_directory> --metrics azure <policyfile>.yml
-
-You can also upload Cloud Custodian logs to App Insights logs::
-
-  $ custodian run --log-group=azure://cloud-custodian/<dev-account>/<region> -s <output_directory> <policyfile>.yml
-
-And you can output logs and resource records to Azure storage accounts::
-
-  $ custodian run -s azure://<my-bucket><my-prefix> <policyfile>.yml
+Cloud Custodian policies can emit logs and metrics to Application Insights after the policy executes.
```
Comment (kapilt) [inline docs/source/azure/gettingstarted.rst]: s/after/when/ - the logs are streamed, metrics are posted as one of the last step of execution, although they will post during if the buffer for them overflows.

**Comment by stefangordon** [comment]
We should probably explain/show how to do this with `execution-options` in the policy for serverless as that is a common use case and this is the page someone will find when they need to do it.

**Comment by aluong** [comment]
With #4316 there will be changes to include the blob output section together with this one, so I will just wait for that change to arrive.

File: docs/source/azure/advanced/appinsightslogging.rst [hunk 2]
```diff
@@ -6,39 +6,66 @@ App Insights Logging & Metrics
...
 
-        custodian run -l azure://<instrumentation_key_guid>
+        custodian run -s <output_directory> -l azure://<instrumentation_key_guid> policy.yml
 
 Usage example using resource name:
 
     .. code-block:: sh
 
-        custodian run -l azure://<resource_group_name>/<app_insights_name>
+        custodian run  -s <output_directory> -l azure://<resource_group_name>/<app_insights_name> policy.yml
```
Comment (stefangordon) [inline docs/source/azure/advanced/appinsightslogging.rst]: nit: is there an extra space between `run` and `-s`?

File: docs/source/azure/advanced/appinsightslogging.rst [hunk 3]
```diff
@@ -6,39 +6,66 @@ App Insights Logging & Metrics
...
 
 
 Writing Custodian Metrics to Azure App Insights
 -----------------------------------------------
 
-By default Cloud Custodian generates App Insights metrics on each
-policy for the number of resources that matched the set of filters,
-the time to retrieve and filter the resources, and the time to execute
-actions.
+By default, Cloud Custodian will upload the following metrics in all mode:
```
Comment (stefangordon) [inline docs/source/azure/advanced/appinsightslogging.rst]: nit: sp: mode**s**

**Comment by kapilt** [comment]
fwiw for aws we've moved from using relying on users having to always specify execution-options on mode to automatically filling execution-options from cli parameters, then users only need to specify execution-options on mode when the value differs from whats on the cli.

File: c7n/ctx.py [hunk 4]
```diff
@@ -47,7 +47,9 @@ def __init__(self, session_factory, policy, options):
         self.sys_stats = None
 
         # A few tests patch on metrics flush
-        self.metrics = metrics_outputs.select(self.options.metrics_enabled, self)
+        # For backward compatibility, accept both 'metrics' and 'metrics_enabled' params
```
Comment (kapilt) [inline c7n/ctx.py]: can you add the pull request number here so we can date this, some day, we'll probably remove on the backwards compatibility.

File: docs/source/quickstart/advanced.rst [hunk 5]
```diff
@@ -161,7 +161,7 @@ is exceeded, policy execution is stopped before taking any actions::
   $ custodian run -s out policy.yml
   custodian.commands:ERROR policy: log-delete exceeded resource limit: 2.5% found: 1 total: 1
 
-If metrics are being published ('-m/--metrics-enabled') then an additional
+If metrics are being published ('-m/--metrics') then an additional
 metric named `ResourceLimitExceeded` will be published with the number
```
Comment (kapilt) [inline docs/source/quickstart/advanced.rst]: ResourceCount  is the actual name of the metric

**Pull request merged**